prboom2/src/p_pspr.c

```diff
@@ -84,14 +84,63 @@ static const int recoil_values[] = {    // phares
   80  // wp_supershotgun
 };
 
+// [XA] keep a list of all parameterized weapon pointers,
+// so we can disable the misc1/misc2 weapon offset
+// behavior for states that use one of these
+
+static const actionf_t param_weapon_ptrs[] = {
+	A_WeaponProjectile,
+	A_WeaponBulletAttack,
+	A_WeaponSound,
+	A_WeaponJump,
+	A_ConsumeAmmo,
+	A_CheckAmmo,
+	A_RefireTo,
+	A_GunFlashTo,
+
+	// This NULL entry must be the last in the list
+	NULL
+};
+
+//
+// P_IsParamWeaponPtr
+// Returns true if the specified action is a
+// parameterized weapon code pointer.
+//
+dboolean P_IsParamWeaponPtr(const actionf_t ptr)
+{
+  int i;
+
+  if (ptr == NULL)
+    return false;
+
+  i = -1;
+  do
+    {
+      ++i;
+      if (param_weapon_ptrs[i] == ptr)
+        return true;
+    }
+  while (param_weapon_ptrs[i] != NULL);
+
+  return false;
+}
+
 //
 // P_SetPsprite
 //
 
 void P_SetPsprite(player_t *player, int position, statenum_t stnum)
```

Comment:
This code isn't something that gets called a lot in one frame, right? Just making sure adding a loop lookup through the weapon pointer array isn't going to affect performance.

Reply:
Nah, should be fine -- there's only ever 2 psprites per player (weapon and flash) -- maybe only 2 ever? I dunno if classic doom MP even creates psprites for the other players. If this was something that applied to actor states, then maybe we'd have a problem. :P